### ISSUE TITLE
Why take 1 away from targets?

### ISSUE THREAD

**Issue by jimanvlad** [CRIPAC-DIG/SR-GNN#6]
Hi,

In your model's loss calculation, you feed the labels in but take 1 away from them:
`tf.nn.sparse_softmax_cross_entropy_with_logits(labels=self.tar - 1, logits=logits))`

Why is that?

**Comment by tangrizzly** [comment]
Because after data preprocessing, we treat 0 as the virtual item to pad the sequence. `self.tar - 1` is a method of compensation to match the target items with the index of scores.
